src/Core/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -100,10 +110,10 @@ public bool TryGetDocument([NotNullWhen(true)] out string? document)
         /// <exception cref="DataApiBuilderException">Raised when document is already generated
         /// or a failure occurs during generation.</exception>
         /// <seealso cref="https://github.com/microsoft/OpenAPI.NET/blob/1.6.3/src/Microsoft.OpenApi/OpenApiSpecVersion.cs"/>
-        public void CreateDocument()
+        public void CreateDocument(bool isHotReloadScenario = false)
```

Comment:
nit: after looking at this with fresh eyes, I think a param name like `doOverrideExistingDocument` may be more appropriate because it self-describes what your intent is. HotReload doesn't technically mean anything inside this `CreateDocument` function. Your added `Documentor_ConfigChangeEventReceived` function is the "hot-reload aware" function and that function just needs to have openApiDocumentor create a new doc even if one may already exist.